pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go

```diff
@@ -341,72 +342,158 @@ func (p *AgentPoller) Stop() {
+			cmdResult, err := con.QemuAgentCommand(`{"execute":"`+string(command)+`"}`, domainName)
 			if err != nil {
-				log.Log.Errorf("Cannot parse guest agent filesystem %s", err.Error())
 				continue
 			}
-			agentStore.Store(GET_FILESYSTEM, filesystems)
-		case GET_AGENT:
+
 			agent, err := parseAgent(cmdResult)
 			if err != nil {
 				log.Log.Errorf("Cannot parse guest agent information %s", err.Error())
 				continue
 			}
+
 			agentStore.Store(GET_AGENT, agent)
 		}
 	}
 }
+
+func getGuestInfo(infoType libvirt.DomainGuestInfoTypes, con cli.Connection, domainName string) (*libvirt.DomainGuestInfo, error) {
+	domain, err := con.LookupDomainByName(domainName)
+	if err != nil {
+		log.Log.Errorf("Failed to lookup domain (%s): %v", domainName, err)
+		return nil, err
+	}
+
+	guestInfo, err := domain.GetGuestInfo(infoType, 0)
+	if err != nil {
+		log.Log.Errorf("Failed to get guest info (%d): %v", infoType, err)
+		return nil, err
+	}
+	return guestInfo, nil
+}
+
+func convertToInterfaces(guestInfo *libvirt.DomainGuestInfo) []api.InterfaceStatus {
+	var interfaceStatuses []api.InterfaceStatus
+	if guestInfo.Interfaces != nil {
+		for _, netInterface := range guestInfo.Interfaces {
+			interfaceStatuses = append(interfaceStatuses, api.InterfaceStatus{
+				Mac:           netInterface.Hwaddr,
+				InterfaceName: netInterface.Name,
+			})
+
+			// TOOD: Extract IP addresses
+		}
+	}
+	return interfaceStatuses
+}
+
+func convertToOSInfo(guestInfo *libvirt.DomainGuestInfo) api.GuestOSInfo {
+	guestInfoOS := api.GuestOSInfo{}
+	if guestInfo.OS != nil {
+		guestInfoOS = api.GuestOSInfo{
+			Name:          guestInfo.OS.Name,
+			KernelRelease: guestInfo.OS.KernelRelease,
+			Version:       guestInfo.OS.Version,
+			PrettyName:    guestInfo.OS.PrettyName,
+			VersionId:     guestInfo.OS.VersionID,
+			KernelVersion: guestInfo.OS.KernelVersion,
+			Machine:       guestInfo.OS.Machine,
+			Id:            guestInfo.OS.ID,
+		}
+	}
+	return guestInfoOS
+}
+
+func convertToTimezone(guestInfo *libvirt.DomainGuestInfo) api.Timezone {
+	timezone := api.Timezone{}
+	if guestInfo.TimeZone != nil {
+		timezone = api.Timezone{
+			Zone:   guestInfo.TimeZone.Name,
+			Offset: guestInfo.TimeZone.Offset,
+		}
+	}
+	return timezone
+}
+
+func convertToUsers(guestInfo *libvirt.DomainGuestInfo) []api.User {
+	var users []api.User
+	if guestInfo.Users != nil {
+		for _, user := range guestInfo.Users {
+			users = append(users, api.User{
+				Name:      user.Name,
+				Domain:    user.Domain,
+				LoginTime: float64(user.LoginTime),
+			})
+		}
+	}
+	return users
+}
+
+func convertToFileSystem(guestInfo *libvirt.DomainGuestInfo) []api.Filesystem {
+	fileSystems := []api.Filesystem{}
+	if guestInfo.FileSystems != nil {
+		for _, filesystem := range guestInfo.FileSystems {
+			disks := []api.FSDisk{}
+			for _, disk := range filesystem.Disks {
+				disks = append(disks, api.FSDisk{
+					Serial: disk.Serial,
+					// BusType: fsDisk.BusType, TODO: BusType is not defined
```

Comment:
I don't see in libvirt API an option to get `BusType`. I'll try to ask about it in #virt Slack channel.

Reply:
`BusType` was added by mistake to KubeVirt API in https://github.com/kubevirt/kubevirt/pull/10970 as there is no usage for it. Also, libvirt by itself (and Go API) doesn't expose BusType.

Reply:
@codingben Hi, since we've GAed the bus type field in KV API we would need to ask libvirt to expose it in libvirt API

Reply:
This likely won't be done in this PR, as it takes time to get another libvirt 11.2 release for it (currently unreleased yet).